src/accelerate/big_modeling.py

```diff
@@ -93,6 +93,13 @@ def init_on_device(device: torch.device, include_buffers: bool = False):
         tst = nn.Liner(100, 100)  # on `cuda` device
     ```
     """
+    if include_buffers and hasattr(device, "__enter__") and callable(getattr(device, "__enter__")):
+        try:
+            device.__enter__()
+            yield
+        finally:
+            return device.__exit__()
```

Comment:
Cool, I didn't know that `device` supports this now.

Instead of entering and exiting explicitly, would it be possible to use it like this?

```python
with device:
    yield
return
```